internal/verifier/verifier.go

```diff
@@ -172,13 +172,13 @@ func (verifier *ProvenanceIRVerifier) Verify() (VerificationResult, error) {
 	}
 
 	// Verify HasBuildCmd.
-	if verifier.Want.WantBuildCmds {
+	if verifier.Got.HasBuildCmd() && verifier.Want.WantBuildCmds {
 		nextResult := verifyHasBuildCmd(verifier.Got)
 		combinedResult.Combine(nextResult)
 	}
 
 	// Verify BuilderImageDigest.
-	if verifier.Want.BuilderImageSHA256Digests != nil {
+	if verifier.Got.HasBuilderImageSHA256Digest() && verifier.Want.BuilderImageSHA256Digests != nil {
```

Comment:
Same as above. I think we need to distinguish between "does not have" and "cannot have".